fence/blueprints/oauth2.py

```
@@ -282,7 +289,20 @@ def get_token(*args, **kwargs):
     See the OpenAPI documentation for detailed specification, and the OAuth2
     tests for examples of some operation and correct behavior.
     """
-    return server.create_token_response()
+    try:
+        response = server.create_token_response()
+    except Exception as e:
```

are there specific exceptions you can catch for known potential issues? I'd feel more comfortable _not_ catching any issues (which may include Python errors we didn't expect)

---

well previously _all_ exceptions were displayed in the html template, now _all_ exceptions return JSON
I think it's fine to catch all exceptions here, because we want this endpoint to always return JSON, even in case of internal error

---

i can log the traceback though so that we can debug if there is a 500 error

---

return JSON in a `finally` clause without catching exception? Not sure--but an idea

---

https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions